var/spack/repos/builtin/packages/fenics/package.py

```diff
@@ -10,26 +10,37 @@ class Fenics(CMakePackage):
     """FEniCS is organized as a collection of interoperable components
     that together form the FEniCS Project. These components include
     the problem-solving environment DOLFIN, the form compiler FFC, the
-    finite element tabulator FIAT, the just-in-time compiler Instant,
-    the code generation interface UFC, the form language UFL and a
-    range of additional components."""
+    finite element tabulator FIAT, the just-in-time compiler Instant / Dijitso,
+    the code generation interface UFC, the form language UFL and a range of
+    additional components."""
 
     homepage = "http://fenicsproject.org/"
+    git      = "https://bitbucket.org/fenics-project/dolfin.git"
     url      = "https://bitbucket.org/fenics-project/dolfin/downloads/dolfin-1.6.0.tar.gz"
     base_url = "https://bitbucket.org/fenics-project/{pkg}/downloads/{pkg}-{version}.tar.gz"
 
-    python_components = ['ufl', 'ffc', 'fiat', 'instant']
+    version('2019.1.0.post0', sha256='61abdcdb13684ba2a3ba4afb7ea6c7907aa0896a46439d3af7e8848483d4392f')
+    version('2018.1.0.post1', sha256='425cc49b90e0f5c2ebdd765ba9934b1ada97e2ac2710d982d6d267a5e2c5982d')
+    version('2017.2.0.post0', sha256='d3c40cd8c1c882f517999c25ea4220adcd01dbb1d829406fce99b1fc40184c82')
+    version('2016.2.0',       sha256='c6760996660a476f77889e11e4a0bc117cc774be0eec777b02a7f01d9ce7f43d')
+    version('1.6.0',          sha256='67f66c39983a8c5a1ba3c0787fa9b9082778bc7227b25c7cad80dc1299e0a201')
 
+    dolfin_versions = ['2019.1.0', '2018.1.0', '2017.2.0', '2016.2.0', '1.6.0']
+
+    variant('python',       default=True,  description='Compile with Python interface')
     variant('hdf5',         default=True,  description='Compile with HDF5')
     variant('parmetis',     default=True,  description='Compile with ParMETIS')
     variant('scotch',       default=True,  description='Compile with Scotch')
     variant('petsc',        default=True,  description='Compile with PETSc')
     variant('slepc',        default=True,  description='Compile with SLEPc')
-    variant('trilinos',     default=True,  description='Compile with Trilinos')
+    variant('py-petsc4py',  default=True,  description='Use PETSC4py')
+    variant('py-slepc4py',  default=True,  description='Use SLEPc4py')
```

Comment:
I think it would be more clear if these variants were called `petsc4py` and `slepc4py`, without the `py-`. Also, is it possible to enable these variants when `~python`, or do we need to add a conflict for that?

Reply:
I agree with your first point to remove the `py-`. The reason for having the `petsc4py` and `slepcy4py` as a variant in there is for later use in the `cmake_args` function (see `self.define_from_variant` functions). Is there another and maybe more elegant way to omit the variants definitions at the beginning and to set the `cmake_args`?

Reply:
There's nothing wrong with the use of variants and `self.define_from_variant`, your current approach is correct.

Reply:
Alright, thanks Adam. But do you know if there is a way to make the `petsc4py` and `slepc4py` variants dependent on the `python` and `petsc` / `slepc` variants to avoid compatibility conflicts?

Reply:
```python
conflicts('+petsc4py', when='~python')
conflicts('+petsc4py', when='~petsc')
conflicts('+slepc4py', when='~python')
conflicts('+slepc4py', when='~slepc')
```